### ISSUE TITLE
aws-ecs/deploy-service-update job clears the task definition app mesh proxy settings

### ISSUE THREAD

**Issue by marcinkepa** [CircleCI-Public/aws-ecs-orb#29]
Task definition before running the job:

```
{
  "taskDefinition": {
    "taskDefinitionArn": "arn:aws:ecs:eu-west-1:xxxxxxxxxxxxx:task-definition/demo-todo:18",
    "containerDefinitions": [
      {
        "name": "demo-todo",
        "image": "xxxxxxxxxxxxx.dkr.ecr.eu-west-1.amazonaws.com/demo/demo-todo-service:783ca12af20f34ff6f06f6408a94f41d4095ae34",
        "cpu": 0,
        "portMappings": [
          {
            "containerPort": 5002,
            "hostPort": 5002,
            "protocol": "tcp"
          }
        ],
        "essential": true,
        "environment": [],
        "mountPoints": [],
        "volumesFrom": [],
        "dependsOn": [
          {
            "containerName": "envoy",
            "condition": "HEALTHY"
          }
        ],
        "logConfiguration": {
          "logDriver": "awslogs",
          "options": {
            "awslogs-group": "/ecs/demo-todo",
            "awslogs-region": "eu-west-1",
            "awslogs-stream-prefix": "ecs"
          }
        },
        "healthCheck": {
          "command": ["CMD-SHELL", "curl -f http://localhost:5002/api/instance || exit 1"],
          "interval": 5,
          "timeout": 5,
          "retries": 3,
          "startPeriod": 5
        }
      },
      {
        "name": "envoy",
        "image": "111345817488.dkr.ecr.us-west-2.amazonaws.com/aws-appmesh-envoy:v1.9.1.0-prod",
        "cpu": 0,
        "memory": 500,
        "portMappings": [],
        "essential": true,
        "environment": [
          {
            "name": "APPMESH_VIRTUAL_NODE_NAME",
            "value": "mesh/demo-api-mesh/virtualNode/demo-todo-vn"
          },
          {
            "name": "ENABLE_ENVOY_XRAY_TRACING",
            "value": "1"
          },
          {
            "name": "ENVOY_LOG_LEVEL",
            "value": "debug"
          }
        ],
        "mountPoints": [],
        "volumesFrom": [],
        "user": "1337",
        "healthCheck": {
          "command": ["CMD-SHELL", "curl -s http://localhost:9901/server_info | grep state | grep -q LIVE"],
          "interval": 5,
          "timeout": 2,
          "retries": 3,
          "startPeriod": 10
        }
      },
      {
        "name": "xray-daemon",
        "image": "amazon/aws-xray-daemon",
        "cpu": 0,
        "portMappings": [
          {
            "containerPort": 2000,
            "hostPort": 2000,
            "protocol": "udp"
          }
        ],
        "essential": true,
        "environment": [],
        "mountPoints": [],
        "volumesFrom": [],
        "dependsOn": [
          {
            "containerName": "envoy",
            "condition": "HEALTHY"
          }
        ],
        "logConfiguration": {
          "logDriver": "awslogs",
          "options": {
            "awslogs-group": "/ecs/demo-todo",
            "awslogs-region": "eu-west-1",
            "awslogs-stream-prefix": "ecs"
          }
        }
      }
    ],
    "family": "demo-todo",
    "taskRoleArn": "arn:aws:iam::xxxxxxxxxxxxx:role/demo-api-test-ecs-task",
    "executionRoleArn": "arn:aws:iam::xxxxxxxxxxxxx:role/ecsTaskExecutionRole",
    "networkMode": "awsvpc",
    "revision": 18,
    "volumes": [],
    "status": "ACTIVE",
    "requiresAttributes": [
      {
        "name": "ecs.capability.execution-role-awslogs"
      },
      {
        "name": "com.amazonaws.ecs.capability.ecr-auth"
      },
      {
        "name": "com.amazonaws.ecs.capability.docker-remote-api.1.17"
      },
      {
        "name": "com.amazonaws.ecs.capability.task-iam-role"
      },
      {
        "name": "ecs.capability.aws-appmesh"
      },
      {
        "name": "ecs.capability.container-health-check"
      },
      {
        "name": "ecs.capability.execution-role-ecr-pull"
      },
      {
        "name": "com.amazonaws.ecs.capability.docker-remote-api.1.18"
      },
      {
        "name": "ecs.capability.task-eni"
      },
      {
        "name": "com.amazonaws.ecs.capability.docker-remote-api.1.29"
      },
      {
        "name": "com.amazonaws.ecs.capability.logging-driver.awslogs"
      },
      {
        "name": "com.amazonaws.ecs.capability.docker-remote-api.1.19"
      },
      {
        "name": "ecs.capability.container-ordering"
      }
    ],
    "placementConstraints": [],
    "compatibilities": ["EC2", "FARGATE"],
    "requiresCompatibilities": ["FARGATE"],
    "cpu": "512",
    "memory": "1024",
    "proxyConfiguration": {
      "type": "APPMESH",
      "containerName": "envoy",
      "properties": [
        {
          "name": "ProxyIngressPort",
          "value": "15000"
        },
        {
          "name": "AppPorts",
          "value": "5002"
        },
        {
          "name": "EgressIgnoredIPs",
          "value": "169.254.170.2,169.254.169.254"
        },
        {
          "name": "IgnoredGID",
          "value": ""
        },
        {
          "name": "EgressIgnoredPorts",
          "value": ""
        },
        {
          "name": "IgnoredUID",
          "value": "1337"
        },
        {
          "name": "ProxyEgressPort",
          "value": "15001"
        }
      ]
    }
  }
}
```

Task definition after running the job:

```
{
  "taskDefinition": {
    "taskDefinitionArn": "arn:aws:ecs:eu-west-1:xxxxxxxxxxxxx:task-definition/demo-todo:19",
    "containerDefinitions": [
      {
        "name": "demo-todo",
        "image": "xxxxxxxxxxxxx.dkr.ecr.eu-west-1.amazonaws.com/demo/demo-todo-service:18afddee681018a172eb3bf52f04dba59bc17b92",
        "cpu": 0,
        "portMappings": [
          {
            "containerPort": 5002,
            "hostPort": 5002,
            "protocol": "tcp"
          }
        ],
        "essential": true,
        "environment": [],
        "mountPoints": [],
        "volumesFrom": [],
        "dependsOn": [
          {
            "containerName": "envoy",
            "condition": "HEALTHY"
          }
        ],
        "logConfiguration": {
          "logDriver": "awslogs",
          "options": {
            "awslogs-group": "/ecs/demo-todo",
            "awslogs-region": "eu-west-1",
            "awslogs-stream-prefix": "ecs"
          }
        },
        "healthCheck": {
          "command": ["CMD-SHELL", "curl -f http://localhost:5002/api/instance || exit 1"],
          "interval": 5,
          "timeout": 5,
          "retries": 3,
          "startPeriod": 5
        }
      },
      {
        "name": "envoy",
        "image": "111345817488.dkr.ecr.us-west-2.amazonaws.com/aws-appmesh-envoy:v1.9.1.0-prod",
        "cpu": 0,
        "memory": 500,
        "portMappings": [],
        "essential": true,
        "environment": [
          {
            "name": "APPMESH_VIRTUAL_NODE_NAME",
            "value": "mesh/demo-api-mesh/virtualNode/demo-todo-vn"
          },
          {
            "name": "ENABLE_ENVOY_XRAY_TRACING",
            "value": "1"
          },
          {
            "name": "ENVOY_LOG_LEVEL",
            "value": "debug"
          }
        ],
        "mountPoints": [],
        "volumesFrom": [],
        "user": "1337",
        "healthCheck": {
          "command": ["CMD-SHELL", "curl -s http://localhost:9901/server_info | grep state | grep -q LIVE"],
          "interval": 5,
          "timeout": 2,
          "retries": 3,
          "startPeriod": 10
        }
      },
      {
        "name": "xray-daemon",
        "image": "amazon/aws-xray-daemon",
        "cpu": 0,
        "portMappings": [
          {
            "containerPort": 2000,
            "hostPort": 2000,
            "protocol": "udp"
          }
        ],
        "essential": true,
        "environment": [],
        "mountPoints": [],
        "volumesFrom": [],
        "dependsOn": [
          {
            "containerName": "envoy",
            "condition": "HEALTHY"
          }
        ],
        "logConfiguration": {
          "logDriver": "awslogs",
          "options": {
            "awslogs-group": "/ecs/demo-todo",
            "awslogs-region": "eu-west-1",
            "awslogs-stream-prefix": "ecs"
          }
        }
      }
    ],
    "family": "demo-todo",
    "taskRoleArn": "arn:aws:iam::xxxxxxxxxxxxx:role/demo-api-test-ecs-task",
    "executionRoleArn": "arn:aws:iam::xxxxxxxxxxxxx:role/ecsTaskExecutionRole",
    "networkMode": "awsvpc",
    "revision": 19,
    "volumes": [],
    "status": "ACTIVE",
    "requiresAttributes": [
      {
        "name": "com.amazonaws.ecs.capability.logging-driver.awslogs"
      },
      {
        "name": "ecs.capability.execution-role-awslogs"
      },
      {
        "name": "com.amazonaws.ecs.capability.ecr-auth"
      },
      {
        "name": "com.amazonaws.ecs.capability.docker-remote-api.1.19"
      },
      {
        "name": "com.amazonaws.ecs.capability.docker-remote-api.1.17"
      },
      {
        "name": "com.amazonaws.ecs.capability.task-iam-role"
      },
      {
        "name": "ecs.capability.container-health-check"
      },
      {
        "name": "ecs.capability.container-ordering"
      },
      {
        "name": "ecs.capability.execution-role-ecr-pull"
      },
      {
        "name": "com.amazonaws.ecs.capability.docker-remote-api.1.18"
      },
      {
        "name": "ecs.capability.task-eni"
      },
      {
        "name": "com.amazonaws.ecs.capability.docker-remote-api.1.29"
      }
    ],
    "placementConstraints": [],
    "compatibilities": ["EC2", "FARGATE"],
    "requiresCompatibilities": ["FARGATE"],
    "cpu": "512",
    "memory": "1024"
  }
}
```

proxyConfiguration is missing after the task definition is being updated by the job.

**Comment by lokst** [comment]
Thanks for the report @marcinkepa, I'll look into this

**Comment by lokst** [comment]
@marcinkepa It would be really helpful if I could have a closer look at the configuration. Would you mind opening a support ticket at https://support.circleci.com/hc/en-us/requests/new , and attaching a copy of the config.yml used and also linking to the failed CircleCI job?

**Comment by marcinkepa** [comment]
@lokst I have opened the ticket (55115)

**Comment by lokst** [comment]
Based on my findings, this is caused by https://github.com/CircleCI-Public/aws-ecs-orb/issues/30.

**Comment by marcinkepa** [comment]
Solved in circleci/aws-ecs@0.0.10
